src/nvim/CMakeLists.txt

```diff
@@ -252,6 +268,17 @@ add_custom_command(OUTPUT ${GENERATED_OPTIONS}
   DEPENDS ${OPTIONS_GENERATOR} ${OPTIONS_LIST_FILE}
 )
 
+# Make NEOVIM_GENERATED_FOR_SOURCES and NEOVIM_GENERATED_FOR_HEADERS mutually exclusive
+# If a header in NEOVIM_GENERATED_FOR_HEADERS is also included in NEOVIM_GENERATED_SOURCES,
+# the header should be included only in NEOVIM_GENERATED_HEADERS
+
+foreach(hfile ${NEOVIM_GENERATED_FOR_HEADERS})
+  list(FIND NEOVIM_GENERATED_FOR_SOURCES ${hfile} hfile_idx)
+  if(NOT ${hfile_idx} EQUAL -1)
+    list(REMOVE_AT NEOVIM_GENERATED_FOR_SOURCES ${hfile_idx})
```

Comment:
> And, if CMake permits duplicated files to build dependency trees, then these two lists don't need to be mutually exclusive.

Need to: yes. But as I said above I know no such files which are included in both. And as I said below I think this is a bad practice in first place. There are also a lot of cases when it is not only bad, it may as well make including header which included generated file in a `*.c` file which also include the same generated file on its own impossible (actually unless it is a generated `*.c.h` file or the like it would be always impossible), at least without taiting the preprocessor. `*.c` file which for some weird reason cannot include some particular header file without one `#undef` or `#define` per identifier defined in that generated file is not something anybody wants in the codebase.